### DISCUSSION TITLE
Pool library

### DISCUSSION THREAD

**Discussion by abensoft-williamthome** [zotonic/zotonic#3394] [Q&A]
Hello!

Sometimes we face some issues with the `z_db_pool` where some connections are not closed properly. We don't know why; we attempt to trace some problems unsuccessfully. Maybe the issue is from Poolboy.
We noticed that Poolboy is not maintained since Jan 2019, and today we saw this [post](https://erlangforums.com/t/pooler-the-most-advanced-erlang-worker-pool-library/2571) about [Pooler](https://github.com/epgsql/pooler):
> Pooler is an advanced worker pool library. It can be used to maintain a pool of, eg, database connections, OS subprocesses, homogenous network connections and so on.
[...] We use it in production in a very critical system to maintain the pool of epgsql PostgreSQL connections.
[...]  Recently it has been transferred to epgsql project, modernized and optimized.

Is there some interest of the Zotonic team in changing the pool library?

**Comment by seannaswell** [comment]
Would migrating to Pooler allow Zotonic to utilize redundant fail-over Postgresql databases?

**Comment by mworrell** [comment]
Definitely a possibility to use `pooler` instead of `poolboy`.
Do you know more people using it?
